frontend/src/components/Navbar.tsx

```diff
@@ -16,14 +16,16 @@ interface IMenuItem {
   link: string;
 }
 
-const navbarItems: Array<IMenuItem> = [{text: 'Home', icon: <HomeIcon/>, link: '/home'},
-                                       {text: 'Reviews', icon: <MenuIcon/>, link: '/reviews'},
-                                       {text: 'Customize', icon: <SettingsIcon/>, link: '/customize'}];
-
-const currPathPrefix = window.location.pathname.split('/')[1]; // Value after first `/`
-const initialIndex = navbarItems.findIndex(({ link }) => link === `/${currPathPrefix}`);
+const navbarItems: Array<IMenuItem> = [
+  { text: 'Home', icon: <HomeIcon />, link: '/home' },
+  { text: 'Reviews', icon: <MenuIcon />, link: '/reviews' },
+  { text: 'Customize', icon: <SettingsIcon />, link: '/customize' },
+];
 
 const Navbar: React.FC = () => {
+  const currPathPrefix = window.location.pathname.split('/')[1]; // Value after first `/`
```

Comment:
Bug: If the navbar was to rerender, it would set the `selectedIndex` value to the `initialIndex` based on the initial entrance site url (first import of `NavBar`), rather than the current url.
Solution: Move the logic which checks the url to the NavBar component so that it is recalculated on each NavBar render